File: backend/src/main/java/com/peakdynamics/service/NewsService.java
package com.peakdynamics.service;

import com.peakdynamics.model.News;
import com.peakdynamics.model.Club;
import java.util.List;

public interface NewsService {
    List<News> getAllNews();
    News getNewsById(Long id);
    List<News> getNewsByClub(Club club);
    List<News> getLatestNews();
}

File: frontend/src/services/newsService.js
import api from './api';

export const newsService = {
  getAllNews: () => api.get('/news'),
  getNewsById: (id) => api.get(`/news/${id}`),
  getNewsByClub: (clubId) => api.get(`/news/club/${clubId}`),
  getLatestNews: () => api.get('/news/latest'),
};